core/src/test/scala/io/qbeast/core/transform/CDFNumericQuantilesTransformationTest.scala

```diff
@@ -24,9 +23,29 @@ class CDFNumericQuantilesTransformationTest extends AnyFlatSpec with Matchers {
     qt.transform(4) should be(1.0)
   }
 
-  it should "return correct transformation for insertion point in middle" in {
-    val qt = CDFNumericQuantilesTransformation(IndexedSeq(1.0, 2.0, 3.0), DoubleDataType)
-    qt.transform(1.5) should be(0.0)
+  it should "return correct transformation for insertion point in the bin" in {
+    val qt = CDFNumericQuantilesTransformation(IndexedSeq(1, 3, 5), orderedDataTypeTest)
+    // 2 is between 1 and 3, so it should be 0.25 (fraction = 2-1 / 3-1 = 0.5. -> 0 + fraction(0.5) / 2 = 0.25)
+    qt.transform(2) should be(0.25)
+  }
+
+  it should "return correct transformation for insertion point in the bin with repeated values" in {
+    val qt = CDFNumericQuantilesTransformation(IndexedSeq(1, 1, 3, 5), orderedDataTypeTest)
+    // 2 is between 1 and 3 so it should be 0.5
+    qt.transform(2) should be(0.5)
+  }
+
+  it should "return correct transformation point for all values inside the bin" in {
+    val quantiles = IndexedSeq(1, 1, 1, 1, 1, 1, 100, 100, 100).map(_.toDouble)
+    val qt =
+      CDFNumericQuantilesTransformation(quantiles, orderedDataTypeTest)
+    val valuesToTest = 2.to(99)
+    val maxIndexQuantiles = quantiles.size - 1
+    valuesToTest.foreach { value =>
```

Comment:
Can you also check that the sequence of transformations is sorted?

Reply:
done!